extensions/jackson/deployment/src/main/java/io/quarkus/jackson/deployment/JacksonProcessor.java

```diff
@@ -395,4 +403,52 @@ public void supportMixins(MixinsRecorder recorder,
                 .supplier(recorder.customizerSupplier(mixinsMap))
                 .done());
     }
+
+    @Record(ExecutionTime.STATIC_INIT)
+    @BuildStep
+    public SyntheticBeanBuildItem jacksonSupport(JacksonSupportRecorder recorder,
+            JacksonBuildTimeConfig jacksonBuildTimeConfig) {
+        return SyntheticBeanBuildItem
+                .configure(JacksonSupport.class)
+                .scope(Singleton.class)
+                .supplier(recorder.supplier(determinePropertyNamingStrategyClassName(jacksonBuildTimeConfig)))
+                .done();
+    }
+
+    private Optional<String> determinePropertyNamingStrategyClassName(JacksonBuildTimeConfig jacksonBuildTimeConfig) {
+        if (jacksonBuildTimeConfig.propertyNamingStrategy.isEmpty()) {
+            return Optional.empty();
+        }
+        var propertyNamingStrategy = jacksonBuildTimeConfig.propertyNamingStrategy.get();
+        try {
+            // let's first try and see if the value is a constant defined in PropertyNamingStrategies
+            Field field = Reflections.findField(PropertyNamingStrategies.class, propertyNamingStrategy);
+            Class<?> clazz = field.get(null).getClass();
+            if (PropertyNamingStrategy.class.isAssignableFrom(clazz)) {
+                return Optional.of(clazz.getName());
+            }
+            throwInvalidPropertyNameStrategyValue(propertyNamingStrategy);
```

Comment:
Do you want this particular exception to be caught by the `catch` block below?

Reply:
It wasn't really correct. It was also not really readable, so I changed things a little